crates/runtime/src/component/dataset/acceleration/on_conflict.rs

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use data_components::util::{column_reference::ColumnReference, on_conflict::OnConflict};
+
+use crate::component::dataset;
+
+use super::{Acceleration, IndexType, OnConflictBehavior};
+
+impl Acceleration {
+    fn on_conflict_targets(&self) -> Vec<ColumnReference> {
+        let mut on_conflict_targets: Vec<ColumnReference> = Vec::new();
+        for (column, index_type) in &self.indexes {
+            if *index_type == IndexType::Unique {
+                on_conflict_targets.push(column.clone());
+            }
+        }
+        if let Some(primary_key) = &self.primary_key {
+            on_conflict_targets.push(primary_key.clone());
+        }
+
+        on_conflict_targets
+    }
+
+    pub fn on_conflict(&self) -> dataset::Result<Option<OnConflict>> {
```

Comment:
This is where we go from the runtime representation of `HashMap<ColumnReference, OnConflictBehavior>` to the type we'll be using from now on, `OnConflict`.

This is where the validation that we enforce only a single on conflict target happens (unless all of the conflict targets are to drop).